test/Java9andUp/src/org/openj9/test/attachAPI/TestJava9AttachAPI.java

```diff
@@ -0,0 +1,106 @@
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.fail;
+
+import java.lang.reflect.Method;
+import java.util.Arrays;
+
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+import org.testng.log4testng.Logger;
+
+/**
+ * This test must be invoked via testng.  Running the main method will return only the status of the attach API.
+ *
+ */
+@Test(groups = { "level.extended" })
+public class TestJava9AttachAPI  {
+	protected static Logger logger = Logger.getLogger(TestJava9AttachAPI.class);
+	private static boolean verbose = Boolean.getBoolean("org.openj9.test.java9AttachAPI.verbose");
+	static final String ATTACH_ENABLE_PROPERTY = "-Dcom.ibm.tools.attach.enable=";
+	static final String ATTACH_SELF_ENABLE_PROPERTY = "-Djdk.attach.allowAttachSelf";
+
+	private String testName;
+	@BeforeMethod(alwaysRun=true)
+	protected void setUp(Method testMethod) throws Exception {
+		testName = testMethod.getName(); 
+		logger.debug("\n------------------------------------------------------\nstarting " + testName);
+		TargetManager.verbose = verbose;
+	}
+
+	@Test
+	public void testAllowAttachSelfDefault() {
+		String targetClass = SelfAttacher.class.getCanonicalName();
+		TargetManager target = new TargetManager(targetClass, null, Arrays.asList(new String[] {ATTACH_ENABLE_PROPERTY}), null);
+		try {
+			int status = target.waitFor();
+			assertEquals(status, ATTACH_SELF_IOEXCEPTION_CODE, "wrong exit status");
+		} catch (InterruptedException e) {
+			fail("unexpected exception", e);
+		}
+	}
+	
+	@Test
+	public void testAllowAttachSelfBareEnable() {
+		String targetClass = SelfAttacher.class.getCanonicalName();
+		TargetManager target = new TargetManager(targetClass, null, 
+				Arrays.asList(new String[] {ATTACH_ENABLE_PROPERTY,  ATTACH_SELF_ENABLE_PROPERTY}), null);
+		try {
+			int status = target.waitFor();
+			assertEquals(status, ATTACH_SELF_API_SUCCEEDED_CODE, "wrong exit status");
+		} catch (InterruptedException e) {
+			fail("unexpected exception", e);
+		}
+	}
+	
+	@Test
+	public void testAllowAttachSelfEnable() {
+		String targetClass = SelfAttacher.class.getCanonicalName();
+		TargetManager target = new TargetManager(targetClass, null, 
+				Arrays.asList(new String[] {ATTACH_ENABLE_PROPERTY,  ATTACH_SELF_ENABLE_PROPERTY+"=true"}), null);
+		try {
+			int status = target.waitFor();
+			assertEquals(status, ATTACH_SELF_API_SUCCEEDED_CODE, "wrong exit status");
+		} catch (InterruptedException e) {
+			fail("unexpected exception", e);
+		}
+	}
+	
+	@Test
+	public void testAllowAttachSelfDisnable() {
+		String targetClass = SelfAttacher.class.getCanonicalName();
+		TargetManager target = new TargetManager(targetClass, null, 
+				Arrays.asList(new String[] {ATTACH_ENABLE_PROPERTY,  ATTACH_SELF_ENABLE_PROPERTY+"=false"}), null);
+		try {
+			int status = target.waitFor();
+			assertEquals(status, ATTACH_SELF_IOEXCEPTION_CODE, "wrong exit status");
```

Comment:
An opportunity for a better error message that wrong exit status (include exit status in the message would aid triage).

Reply:
assertEquals() already does that, e.g.
wrong exit status expected [2] but found [1]